File: AoC2016/JavaScript/day2.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('../day2in.txt','utf8');
input = input.split(/\r?\n/);

//do stuff
const pad = [
            [0,0,'1'],[1,0,'2'],[2,0,'3'],
            [0,1,'4'],[1,1,'5'],[2,1,'6'],
            [0,2,'7'],[1,2,'8'],[2,2,'9']
            ];
var ans = '';

for (i in input)
{
    var coords = [1,1];
    for (j in input[i])
    {
        let curr = input[i][j];
        if ( (curr == 'L' && coords[0] > 0)
            || (curr == 'R' && coords[0] < 2) )
        {
            curr == 'L' ? coords[0]-- : coords[0]++;
        }
        if ( (curr == 'U' && coords[1] > 0)
            || (curr == 'D' && coords[1] < 2) )
        {
            curr == 'U' ? coords[1]-- : coords[1]++;
        }
    }
    for (i in pad)
    {
        if (pad[i][0] == coords[0] && pad[i][1] == coords[1]) ans+=pad[i][2];
    }
}
console.log(ans);

File: AoC2020/Python3/day2.py
#puzzle setup
with open('./day2in.txt','r') as pInput:
  pInput = pInput.read().splitlines()

#do stuff
ans = 0

for i in range(0,len(pInput)):
  parts = pInput[i].split(' ')
  counts = parts[0].split('-')
  low = int(counts[0])
  high = int(counts[1])
  tar = parts[1][0]
  pw = parts[2]
  ct = pw.count(tar)
  
  if ct >= low and ct <= high:
    ans += 1

print(ans)